runtests.py

```diff
@@ -5,49 +5,57 @@
 #
 
 """
-File for running tests programmatically.
+Script for running Spyder tests programmatically.
 """
 
 # Standard library imports
 import os
 import sys
+import argparse
 
 # Third party imports
-import qtpy  # to ensure that Qt4 uses API v2
 import pytest
 
 
 # To run our slow tests only in our CIs
-run_slow = False
-if os.environ.get('CI', None) is not None or '--run-slow' in sys.argv:
-    run_slow = True
+RUN_CI = os.environ.get('CI', None) is not None
+run_slow = RUN_CI
 
-def main():
+
+def main(run_slow=run_slow, extra_args=None):
     """
-    Run pytest tests.
+    Run pytest tests for Spyder.
     """
     pytest_args = ['spyder',
                    'spyder_profiler',
-                   '-x',
                    '-vv',
                    '-rw',
-                   '--durations=10',
-                   '--cov=spyder',
-                   '--cov=spyder_profiler',
-                   '--cov-report=term-missing']
+                   '--durations=10']
 
-    if run_slow:
+    if RUN_CI:
+        pytest_args.append('-x')
+    elif extra_args:
+        pytest_args += extra_args
+    if run_slow and '--run-slow' not in pytest_args:
         pytest_args.append('--run-slow')
 
+    print("Pytest Arguments: " + str(pytest_args))
     errno = pytest.main(pytest_args)
 
-    # sys.exit doesn't work here because some things could be running
-    # in the background (e.g. closing the main window) when this point
-    # is reached. And if that's the case, sys.exit does't stop the
-    # script (as you would expected).
+    # sys.exit doesn't work here because some things could be running in the
+    # background (e.g. closing the main window) when this point is reached.
+    # If that's the case, sys.exit doesn't stop the script as you would expect.
     if errno != 0:
         raise SystemExit(errno)
 
 
 if __name__ == '__main__':
-    main()
+    test_parser = argparse.ArgumentParser(
+        usage='python runtests.py [--run-slow] [-- pytest_args]')
+    test_parser.add_argument('--run-slow', action='store_true',
+                             default=run_slow,
+                             help='Run the slow (mainly GUI/functional) tests')
```

Comment:
There's no need to have the `--run-slow` option defined here. It's handled in `conftest.py`, as the [Pytest docs](https://docs.pytest.org/en/latest/example/simple.html) recommend to do it:

https://github.com/spyder-ide/spyder/blob/3.x/conftest.py#L30

Reply:
The reason I kept it was so it'd be exposed to the user e.g. through ``runtests.py --help`` and have clearer error handling with ``argparse``, etc. However, I can ahead and eliminate it since it does make the code cleaner and it isn't that big of a deal.

Reply:
Yes, please remove it, but leave it in the `usage` text above. I think that's enough.

Reply:
Okay, sure; done.